pilot/pkg/model/push_context.go

```diff
@@ -1061,3 +1063,43 @@ func (ps *PushContext) EnvoyFilters(proxy *Proxy) []*EnvoyFilterWrapper {
 	}
 	return out
 }
+
+// pre computes gateways per namespace
+func (ps *PushContext) initGateways(env *Environment) error {
```

Comment:
Seems like this is never called?

We should skip it if ScopeGatewayToNamespace is disabled though

Reply:
Oh crap. good catch